### DISCUSSION TITLE
How to set a value of a placeholder field/ insert a value

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2273] [Help] [answered]
Hi all,

I do have two fields holding two numeric values.
Depending on setting those values, I want to have a third field hydrated with the product of the first two.
So setting the product field like this:
```
                    BelongsToSelect::make('value1')
                        ->relationship('value1', 'name')
                        ->required()
                        ->reactive()
                        ->afterStateUpdated(function (Closure $get, Closure $set, $state) {;
                            $value2= Value2Model::where('id', $get('Value2'))->first()->value;
                            $value1= Value1Model::where('value', $state)->first()->value;

                            // If both values are given, set the product
                            if ($value1&& $value2) {
                                $set('product', $value1* $value2);
                            }
                        }),
```

And this works just fine using a TextInput::make('product') field.
Now, I would like to have the value set in a Placeholder, not only a "disabled TextInput".

But how to approach that? How to set a value inside a content() of a Placeholder?
Preferably, it would be possible to do the following:
- have a HTMLString in content()
- insert the set value at a certain spot in content()
- being able to set extraAttributes (css classes) depending on the set value

Am I dreaming a bit too much here or is this possible?

Thanks!

**Comment by danharrin** [comment, marked as answer]
You can pass a closure to `content()` and use `$get` to retrieve the data and calculate live, since you cant `$set` a placeholder.

**Reply by vv-ac** [reply]
That worked, thanks!

**Reply by iammuttaqi** [reply]
@danharrin, is there a way to retrieve the value from a placeholder input? Currently, its value isn't accessible using `$this->form->getState()`
